src/Core/src/Hosting/Dispatching/AppHostBuilderExtensions.cs

```diff
@@ -10,7 +10,11 @@ public static partial class AppHostBuilderExtensions
 	{
 		public static MauiAppBuilder ConfigureDispatching(this MauiAppBuilder builder)
 		{
-			builder.Services.TryAddSingleton<IDispatcherProvider>(svc => new DispatcherProvider());
+			builder.Services.TryAddSingleton<IDispatcherProvider>(svc =>
+				// the DispatcherProvider might have already been initialized, so ensure that we are grabbing the
+				// Current and putting it in the DI container.
+				DispatcherProvider.Current);
```

Comment:
If there is no current... Can Current be null? If it's null, is the behaviour to then create a new instance with the default ctor?

Reply:
If it is `null`, then Current will create one:

https://github.com/dotnet/maui/blob/29e841af968ec2999fa6f1294169f964d024f4a6/src/Core/src/Dispatching/DispatcherProvider.cs#L14-L15